spec/features/visitors/sign_up_spec.rb

```diff
@@ -281,4 +255,50 @@
       expect(current_url).to eq new_user_session_url
     end
   end
+
+  context 'user signs up twice without confirming email' do
+    it 'sends the user the confirmation instructions email again' do
+      email = 'test@example.com'
+
+      expect { sign_up_with(email) }.
+        to change { ActionMailer::Base.deliveries.count }.by(1)
+      expect(last_email.html_part.body).to have_content(
+        t('devise.mailer.confirmation_instructions.subject')
+      )
+
+      expect { sign_up_with(email) }.
+        to change { ActionMailer::Base.deliveries.count }.by(1)
+      expect(last_email.html_part.body).to have_content(
+        t('devise.mailer.confirmation_instructions.subject')
+      )
+    end
+  end
+
+  context 'user signs up and requests confirmation email again' do
+    it 'sends the confirmation email again' do
+      sign_up_with('test@example.com')
+      click_on t('links.resend')
+      fill_in :user_email, with: 'test@example.com'
+
+      expect { click_on t('forms.buttons.resend_confirmation') }.
+        to change { ActionMailer::Base.deliveries.count }.by(1)
+
+      expect(last_email.html_part.body).to have_content(
+        t('devise.mailer.confirmation_instructions.subject')
+      )
+    end
+  end
+
+  context 'user signs up and confirms email, goes through flow again' do
```

Comment:
It might be worth pointing out that in this scenario, the user also set a password, because you made a change recently where if they haven't set their password, they will get the confirmation email, not the "you have already signed up" email.